protoc-gen-swagger/genswagger/template.go

```diff
@@ -944,9 +944,16 @@ func renderServices(services []*descriptor.Service, paths swaggerPathsObject, re
 					}
 					if opts.Responses != nil {
 						for name, resp := range opts.Responses {
-							respObj := swaggerResponseObject{
-								Description: resp.Description,
-								Schema:      swaggerSchemaFromProtoSchema(resp.Schema, reg, customRefs, meth),
+							// Merge response data into default response if available.
```

Comment:
This is a potential point of discussion. It's a very slight behavior change for existing protos which set a code 200 response object - doing so previously would result in a missing description (instead of method-comment-based description) and/or return schema (instead of return-proto-based schema), unless each of those fields were manually specified in the response object.

It's very convenient to merge the response specification into the default 200 spec generated from the proto return value + comments. For the case where you just want to customize one field, you don't have to re-specify all the others.

Reply:
Thanks for raising this - I don't think we're too worried about breaking users here, it seems to me a generally better solution :+1: